### ISSUE TITLE
Suggestion : Map UI revamp

### ISSUE THREAD

**Issue by Ouaz** [yairm210/Unciv#6989]
Hello,

I think the map UI could be slightly revamped, to get things more "tidy" and also to gain some visible screen space.

![Map_UI_revamp_n](https://user-images.githubusercontent.com/11946570/170791265-36c4c617-813c-4db5-96bb-7a9aa214b1ff.png)

1) On upper left corner:
- Move the "_Policy_" button under the "_Tech_" button.
- Replace "_Diplomacy_" text by the [diplomacy icon](https://raw.githubusercontent.com/yairm210/Unciv/master/android/Images/OtherIcons/DiplomacyW.png) already in the game, and place the button to the right of the "_Policy_" button.

2) On upper right corner:
- Under "_Next Unit/Turn_" button, add a "_Situation Reports -/+_" toggle button, that allows to show/hide the notification list.
- _Optional_: add the count of notifications for a turn, e.g. "_Situation reports (11) -/+_"

3) On lower right corner:
- Move the 4 icon filter buttons above the mini-map, horizontally.
- Move the tile info panel to the left of the mini-map.
-> This would allow to gain a lot of space for the notification list, and would prevent to squash/overlay it when the tile info panel is big and/or displayed.

_With notification list displayed:_
![Map_UI_revamp_1](https://user-images.githubusercontent.com/11946570/170792879-5858b5ae-7f45-4af4-baa7-d29943ea4af1.png)

As usual, I have no idea if it's doable and if it would require too much work. ^^

Thanks for reading.

**Comment by SomeTroglodyte** [comment]
> Thanks for reading

Thanks for writing. Good bye.

<br /><br /><br /><br /><br /><br />

<details><summary> :upside_down_face: :zany_face: :rofl:  </summary>

Look at that unmerged UI PR - where is it - #6945 - which deals with stuff overlapping on the topbar. When it pushes its two "buttons" down there's the side effect that top right might look uglier as the next turn button can be far wider than the Overview button... I had no idea how to improve that, so I left it. Same on the left -> your idea fits perfectly.

Switching Tileinfo and map toggles: less nice - TileInfo height varies _wildly_ and once it's higher than the minimap - ugly. Less ugly now when it's wider methinks. Also, giant BattleTable would surely overlap.

Notifications - I had the simple idea not yet tested -> Allow it to scroll far far up until almost all are hidden. At the moment it calculates a "virtual empty space" needed to have it extend _under_ tile info and still be able to sctroll the last one into view... (I _thought_ that looked cool) - that space just needs to be a bit larger. Just one number. Problem - when it's gone what to grab, so the calculation must hit the "a bit left"... Also, think CTP - I think that was one of the Civs that allowed you to mark individual messages as read - no, MOO3 did it best! Idea also just simmering around. Or - wrap in expander -> problem: can't leave the ragged left edge transparent to clicks. Or... Yea a button would help but I dislike because - hmm, why indeed... similar _concept_ as an expander but can't simply use it. Maybe I should extend the Widget's abilities...
</details>

**Comment by Ouaz** [comment]
Haha, thanks for the answer!

> switching Tileinfo and map toggles: less nice - TileInfo height varies wildly and once it's higher than the minimap - ugly. Less ugly now when it's wider methinks.

Well, even if the player likes to play with a lilliputian mini-map, the probability that the Tile Info panel gets way higher than the mini-map is low, _I think_. AFAIK, max possible lines is around 10, and even with the minimum map size (after that size step, it disappears):
<details>
<summary>Screenshot</summary>
<img src="https://user-images.githubusercontent.com/11946570/170894725-a7dfa52d-30cc-4b73-8fa4-d645d03199af.png">
</details>

But anyway, suggestion n°3 can be dumped because...

> Also, giant BattleTable would surely overlap.

Sadly yes. :(
<details>
<summary>Screenshot</summary>
<img src="https://user-images.githubusercontent.com/11946570/170894097-23177ac6-e295-423c-902e-670756152aad.png">
</details>

> Look at that unmerged UI PR - where is it - https://github.com/yairm210/Unciv/pull/6945 - which deals with stuff overlapping on the topbar. When it pushes its two "buttons" down there's the side effect that top right might look uglier as the next turn button can be far wider than the Overview button... I had no idea how to improve that, so I left it. Same on the left -> your idea fits perfectly.

That looks good... Well, I didn't plan to suggest this here but...

... I made a suggestion a few weeks ago on Discord that would even fit better with your PR. _I think_ it didn't receive so much enthusiasm, but there's still hope, who knows ^^:

- Move all buttons to the right + replace diplomacy button.
- Move the notification list to the left (aligning icons+texts to the left, obviously) + show/hide toggle button.
Basically a switch. ^^

This would give plenty of space for the notification list (even when unit action buttons are displayed), and all useful buttons are on the same area for the player.
No problem anymore with the different sizes of the mini-map and the Tile Info panel, and also the notification list size/scrolling.
Buttons from the upper bar being pushed down with your PR would look good too.

![UI_switch](https://user-images.githubusercontent.com/11946570/170896260-ff0d56ec-ae60-4df5-80b1-dd9bcf75bf2c.png)

Opinions?

<details>
<summary>!</summary>
Thanks for reading.
</details>

**Comment by SomeTroglodyte** [comment]
> lilliputian 

Size relations also depend on "resolution" choice and map size..

> that it can't work

Re toggle for notifications? Did I say can't? Nah, can but don't like because I can't see a "path" I like. As in coding possible and result pleasing. A "+/-" _button_ would likely go pretty straightforward. Waitaminute - I see something I _would_ like but that's a little challenge...

**Comment by touhidurrr** [comment]
Personally I would like to see the chat button (or diplomacy button maybe) on left and policy on right.
Since already developed the habit of pressing policy button on right and changing this now feels like something is off.

**Comment by Ouaz** [comment]
> Since pretty sure mostly everyone have already developed the habit of pressing policy button on right and changing this now feels like something is off for me.

I thought of it when doing the mockup. Then I thought "Hey, I use the Diplomacy button way more than the Policy button. I think it's better to have it directly available on the lower-right end of the button group, and the Policy button on the screen border".

**Comment by touhidurrr** [comment]
>  I use the Diplomacy button way more than the Policy button.

Well if that was your thought than I think It's not plausible after all. Cause it’s easier to press the diplomacy button if it is at the right side. You just need to use a left hand finger.

So, it should be on left after all.

**Comment by AdityaMH** [comment]
decrease width damage calculation box?

**Comment by Ouaz** [comment]
Most of these UI suggestions have been added. It could be closed.
(battle table overhaul excepted :trollface: )

**Comment by SeventhM** [comment]
> Most of these UI suggestions have been added. It could be closed.
If you say so. Closing then


> the alt battle table still exists

Different issue. I don't think that's necessary for keeping this issue open, no? The only think I would probably want better positioned is a way to access the victory screen, as it can be impossible to access as Maya. But again, there's separate issues for that
